src/content/blog/content-layer-deep-dive.mdx

```diff
@@ -0,0 +1,230 @@
+			const lastModified = meta.get('last-modified');
+
+			// If so, make a conditional request for the feed
+			const headers = lastModified ? { 'If-Modified-Since': lastModified } : {};
+
+			const res = await fetch(feedUrl, { headers });
+
+			// If the feed hasn't changed, you do not need to update the store
+			if (res.status === 304) {
+				logger.info('Feed not modified, skipping');
+				return;
+			}
+			if (!res.ok || !res.body) {
+				throw new Error(`Failed to fetch feed: ${res.statusText}`);
+			}
+
+			// Store the last-modified header in the meta store so we can
+			// send it with the next request
+			meta.set('last-modified', res.headers.get('last-modified'));
+
+			const feed = parseFeed(res.body);
+
+			// If the loader doesn't handle incremental updates, clear the store before inserting new entries
+			// In some cases the API might send a stream of updates, in which case you would not want to clear the store
+			// and instead add, delete or update entries as needed.
+			store.clear();
+
+			for (const item of feed.items) {
+				// The parseData helper uses the schema to validate and transform data
+				const data = await parseData({
+					id: item.guid,
+					data: item,
+				});
+
+				// The generateDigest helper lets you generate a digest based on the content. This is an optional
+				// optimization. When inserting data into the store, if the digest is provided then the store will
+				// check if the content has changed before updating the entry. This will avoid triggering a rebuild
+				// in development if the content has not changed.
+				const digest = generateDigest(data);
+
+				store.set({
+					id,
+					data,
+					// If the data source provides HTML, it can be set in the `rendered` property
+					// This will allow users to use the `<Content />` component in their pages to render the HTML.
+					rendered: {
+						html: data.description ?? '',
+					},
+					digest,
+				});
+			}
+		},
+	};
+}
+```
+
+## When _not_ to use content collections
+
+Previously it was clear when content collections were a good idea, because it was whenever you were using local content in your pages. The content layer API makes that less clear, because you can use it for any content source including live APIs. The key consideration when deciding is that the data is only updated when the site is built. This means it's perfect for prerendered pages, but also for [on-demand rendering (SSR)](https://docs.astro.build/en/guides/server-side-rendering/) where the data changes relatively rarely. A blog would be a good example - if you're writing a blog that's hosted in a CMS, you can trigger a build with a webhook whenever you publish a new post. The same applies for e-commerce, where a build can be triggered when a product is edited. If you are ok with waiting the time it takes to deploy the site to see updates, then content collections are a good choice. You will get the best performance and great developer experience.
```

Comment:
```suggestion
Previously it was clear when content collections were a good idea: whenever you were using local content in your pages! The Content Layer API gives you so much more power and flexibility because you can use it for any content source including live APIs. However, it's important to remember that data is only updated when the site is built, so it won't serve every use case.

This means that collections are perfect when your data changes relatively infrequently, such as a blog. If you're writing a blog that's hosted in a CMS, you can trigger a build with a webhook whenever you publish a new post.

The same applies for most e-commerce sites, where a build can be triggered when a product is edited. If you are ok with waiting the time it takes to deploy the site to publish updates, then content collections are still your obvious choice. You will get the best performance and great developer experience.
```